nir/ir.py

```diff
@@ -7,20 +7,20 @@
 Edges = typing.NewType("Edges", typing.List[typing.Tuple[int, int]])
 
 
-@dataclass
-class NIR:
-    """Neural Intermediate Representation (NIR)
+class NIRNode:
+    """Basic Neural Intermediate Representation Unit (NIRUnit)"""
```

Comment:
why are you using NIRUnit here?

Reply:
Nicely spotted. I blame GitHub copilot :-1: